curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/643c13b3c7e00e01c0e73d52.md

```diff
@@ -0,0 +1,241 @@
+              <th id="avatars">Avatars</th>
+              <th id="replies">Replies</th>
+              <th id="views">Views</th>
+              <th id="activity">Activity</th>
+            </tr>
+          </thead>
+          <tbody id="posts-container"></tbody>
+        </table>
+      </div>
+    </main>
+    <script src="./script.js"></script>
+  </body>
+</html>
+```
+
+```css
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+:root {
+  --main-bg-color: #2a2a40;
+  --black: #000;
+  --dark-navy: #0a0a23;
+  --dark-grey: #d0d0d5;
+  --medium-grey: #dfdfe2;
+  --light-grey: #f5f6f7;
+  --peach: #f28373;
+  --salmon-color: #f0aea9;
+  --light-blue: #8bd9f6;
+  --light-orange: #f8b172;
+  --light-green: #93cb5b;
+  --golden-yellow: #f1ba33;
+  --gold: #f9aa23;
+  --green: #6bca6b;
+}
+
+body {
+  background-color: var(--main-bg-color);
+}
+
+nav {
+  background-color: var(--dark-navy);
+  padding: 10px 0;
+}
+
+.fcc-logo {
+  width: 210px;
+  display: block;
+  margin: auto;
+}
+
+.title {
+  margin: 25px 0;
+  text-align: center;
+  color: var(--light-grey);
+}
+
+.table-wrapper {
+  padding: 0 25px;
+  overflow-x: auto;
+}
+
+table {
+  width: 100%;
+  color: var(--dark-grey);
+  margin: auto;
+  table-layout: fixed;
+  border-collapse: collapse;
+  overflow-x: scroll;
+}
+
+#topics {
+  text-align: start;
+  width: 60%;
+}
+
+th {
+  border-bottom: 2px solid var(--dark-grey);
+  padding-bottom: 10px;
+  font-size: 1.3rem;
+}
+
+td:not(:first-child) {
+  text-align: center;
+}
+
+td {
+  border-bottom: 1px solid var(--dark-grey);
+  padding: 20px 0;
+}
+
+.post-title {
+  font-size: 1.2rem;
+  color: var(--medium-grey);
+  text-decoration: none;
+}
+
+.category {
+  padding: 3px;
+  color: var(--black);
+  text-decoration: none;
+  display: block;
+  width: fit-content;
+  margin: 10px 0 10px;
+}
+
+.career {
+  background-color: var(--salmon-color);
+}
+
+.feedback,
+.html-css {
+  background-color: var(--light-blue);
+}
+
+.support {
+  background-color: var(--light-orange);
+}
+
+.general {
+  background-color: var(--light-green);
+}
+
+.javascript {
+  background-color: var(--golden-yellow);
+}
+
+.backend {
+  background-color: var(--gold);
+}
+
+.python {
+  background-color: var(--green);
+}
+
+.motivation {
+  background-color: var(--peach);
+}
+
+.avatar-container {
+  display: flex;
+  justify-content: center;
+  gap: 10px;
+  flex-wrap: wrap;
+}
+
+.avatar-container img {
+  width: 30px;
+  height: 30px;
+}
+
+@media (max-width: 750px) {
+  .table-wrapper {
+    padding: 0 15px;
+  }
+
+  table {
+    width: 700px;
+  }
+
+  th {
+    font-size: 1.2rem;
+  }
+
+  .post-title {
+    font-size: 1.1rem;
+  }
+}
+```
+
+```js
+const FORUM_LATEST = "https://forum-proxy.freecodecamp.rocks/latest";
+const FORUM_TOPIC_URL = "https://forum.freecodecamp.org/t/";
+const FORUM_CATEGORY_URL = "https://forum.freecodecamp.org/c/";
+const AVATAR_URL = "https://sea1.discourse-cdn.com/freecodecamp";
```

Comment:
we are changing all instances for all steps to use camel case for consistency

add a new step to have campers create four `const` variables using camel case

you can add the new step before this one to have them create these variables. 

Then use a search and replace to replace all instances of the snake case in the seed code to use the camel case
this includes the final solution at the end too 👍

Reply:
@jdwilkin4 need some clarity here please ? before this step, we will tell campers to add 4 const, then do we also tell them to put the values or we add them automatically on next step ?

Reply:
Since we are using `const` we have to declare the variables and assign it a value all in one step. 

If not, an error will be thrown if you declare a variable and don't assign it a value using `const`. So you technically wouldn't be able to go to the next step because editor wouldn't allow it with the error.

Reply:
> we add them automatically on next step

We want to avoid automatically providing code for them unless there is a really good reason to do so.
In this case, it is not necessary to provide the answer for them because this is the last project in the curriculum and campers will have plenty of experience working with variables at this point 👍